requests-for-comments/2024-02-26-user-agent-parsing.md

```diff
@@ -0,0 +1,41 @@
+# Request for comments: User Agent Parsing & Paul D'Ambra
+
+## Problem statement
+
+_Who are we building for, what are their needs (include example cases), why is this important?_
+
+-   User agent parsing is ~3% of the posthog-js bundle and needs increasing code golf to keep under control.
+-   It is also duplicated in posthog-js-lite and needs people to remember to update one when they change the other
+-   Because it is in the SDK other SDKs cannot take advantage of it and so we have things like the user agent populator plugin
+-   Any changes are slow to roll out because of the long-tail of SDK versions in the wild
+
+## Success criteria
+
+_How do we know if this is successful (i.e. metrics, customer feedback), what's out of scope, whats makes this ambitious?_
+
+-   The posthog-js bundle size is reduced by 3% or more
+-   The user agent parsing is available universally across all SDKs
+
+## Design
+
+_What are the key user experience and technical design decisions / trade-offs?_
+
+### Option 1 - do nothing
+
+It's been fine for 2 years, we copy the changes from https://github.com/PostHog/posthog-js/pull/1038 to posthog-js-lite and move on.
+Fast, but nothing is easier next time we need to change this
+
+### Option 2 - move the code into the plugin server
+
+e.g. in this location https://github.com/PostHog/posthog/pull/20548 (but not that implementation)
```

Comment:
would need guidance from team pipeline on where the code should live in the plugin server (or for us to implement it - and then change it - to help them decide)